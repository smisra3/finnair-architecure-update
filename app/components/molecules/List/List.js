// @flow
import type { Map } from 'react';
import { PureComponent } from 'react';
import { fromJS } from 'immutable';

import styled from 'styled-components';
import styles from './List.style';

type Props = {
  className: string,
  listType?: string,
  inheritedStyles?: string,
  listItemStyle?: Map,
  listItems: Map,
  defaultStyles?: boolean,
  elementRef?: any => void,
};

class List extends PureComponent<Props> {
  render() {
    const {
      className,
      listType,
      listItems,
      listItemStyle,
    } = this.props;
    const TagTypeKey = listType ? listType : 'ul';
    const listItemsData = fromJS(listItems);
    const listItemStyleData = fromJS(listItemStyle);
    <TagTypeKey className={className}>
      {listItemsData.map((item, index) => {
        return (<li className={listItemStyleData[index]} keys={`list-index-${index}`}>{item}</li>);
      })}
    </TagTypeKey>
  }
}

List.defaultProps = {
  listType: 'ul',
  inheritedStyles: '',
  listItemStyle: null,
  defaultStyles: false,
  elementRef: null,
};

export default styled(List)`${styles}`;
export { List as ListVanilla };
