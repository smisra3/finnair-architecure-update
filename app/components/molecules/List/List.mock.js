import { action } from '@storybook/addon-actions';

const unorderedList = {
  listType: 'ul',
  inheritedStyles: '',
  listItemStyle: '',
  defaultStyles: false,
  className: 'ul-list',
};

const orderedList = {
  listType: 'ol',
  inheritedStyles: '',
  listItemStyle: '',
  defaultStyles: false,
  className: 'ol-list',
};

export {
  unorderedList,
  orderedList,
};
