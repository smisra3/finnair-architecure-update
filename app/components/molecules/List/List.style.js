import { css } from 'styled-components';

const ListStyles = css`
  list-style-type: none;
  margin: 0;
  padding: 0;
  ${props =>
    (props.defaultStyles && props.listType === 'ul'
      ? `
      list-style-type: circle;
      ` : '')};
  ${props =>
    (props.defaultStyles && props.listType === 'ol'
      ? `
      list-style-type: upper-roman;
      ` : '')};
`;

export default ListStyles;
