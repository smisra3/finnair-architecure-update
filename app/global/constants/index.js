/* TODO: Remove lint exception */
/* eslint-disable import/prefer-default-export */
export const DEVICE_TYPE = 'DEVICE_TYPE';
export const IS_TABLET = 'IS_TABLET';
export const REGION_TYPE = 'REGION_TYPE';
export const CONFIG_KEYS = 'CONFIG_KEYS';
export const CURRENT_ROUTE = 'CURRENT_ROUTE';
export const PAGE_URL = 'PAGE_URL';
export const PAGE_QUERY = 'PAGE_QUERY';
export const GLOBAL_DATA_FAILURE = 'FIL/GLOBAL/GLOBAL_DATA_FAILURE';
export const CHECK_AUTH = 'FIL/GLOBAL/CHECK_AUTH';
export const CHECK_AUTH_SUCCESS = 'FIL/GLOBAL/CHECK_AUTH_SUCCESS';
export const SET_USER_STATE = 'FIL/GLOBAL/SET_USER_STATE';
export const PAGE_ORIGIN = 'PAGE_ORIGIN';
export const LABELS_STATE_KEY = 'labels';

export const USER_STATE_LOGGED_IN = 'FIL/GLOBAL/SUCCESS';
export const USER_STATE_RECOGNIZED = 'FIL/GLOBAL/RECOGNIZED';
export const USER_STATE_ANONYMOUS = 'FIL/GLOBAL/ANONYMOUS';

export const GET_LABELS = 'FIL/GLOBAL/GET_LABELS';
export const GET_LABELS_SUCCESS = 'FIL/GLOBAL/GET_LABELS_SUCCESS';
export const GET_LABELS_FAILURE = 'FIL/GLOBAL/GET_LABELS_FAILURE';
