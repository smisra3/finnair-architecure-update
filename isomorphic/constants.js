const HEARTBEAT_SESSION_NAME = 'JSESSIONID';
const ENV_DEVELOPMENT = 'development';
const ENV_PRODUCTION = 'production';
const SERVICE_WORKER_FILE = '/service-worker.js';
const API_PROXY_PATH = '/api';
const CLEAR_SERVICES_CACHE = '/clear-services-cache';
const HEALTH_CHECK = '/healthcheck';
const PROTOCOL_HTTPS = 'https';
const RESPONSE_OK = 200;
const RESPONSE_INTERNAL_SERVER_ERROR = 500;
const STATIC_FOLDER_PREFIX = '/static/';

module.exports = {
  HEARTBEAT_SESSION_NAME,
  ENV_DEVELOPMENT,
  ENV_PRODUCTION,
  SERVICE_WORKER_FILE,
  API_PROXY_PATH,
  CLEAR_SERVICES_CACHE,
  PROTOCOL_HTTPS,
  RESPONSE_OK,
  RESPONSE_INTERNAL_SERVER_ERROR,
  STATIC_FOLDER_PREFIX,
  HEALTH_CHECK,
};
