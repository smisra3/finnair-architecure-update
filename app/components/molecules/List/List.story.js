// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  unorderedList,
  orderedList,
} from './List.mock';

// Import Vanilla Component to display markup and props
import { ListVanilla } from './List';
// Import Styled Component to showcase variations
import List from './index';

storiesOf('Molecules', module).addWithChapters('List', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ListVanilla {...unorderedList} className={`hide-default-sample ${primaryButton.className}`} />,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'List Variations',
      sections: [
        {
          title: 'Unordered List',
          sectionFn: () => <List {...unorderedList} />,
        },
        {
          title: 'Ordered List',
          sectionFn: () => <List {...orderedList} />,
        },
      ],
    },
  ],
});
