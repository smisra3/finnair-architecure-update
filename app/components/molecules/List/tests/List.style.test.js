import { shallow } from 'enzyme';
import 'jest-styled-components';

import List from '..';
import Theme from '../../../../styles/theme';
import { DESKTOP, MOBILE } from '../../../../constants';

describe('List', () => {
  let list = '';
  const theme = { ...Theme };

  test('should render correctly', () => {
    theme.deviceType = DESKTOP;
    list = shallow(<List theme={theme} />);
    expect(list).toMatchSnapshot();
  });

  test('should render correctly for mobile', () => {
    theme.deviceType = MOBILE;
    list = shallow(<List theme={theme} />);
    expect(list).toMatchSnapshot();
  });
});
