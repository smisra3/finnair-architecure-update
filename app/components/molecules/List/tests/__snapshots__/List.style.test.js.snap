// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
.c0 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

<List
  className="c0"
  defaultStyles={false}
  elementRef={null}
  inheritedStyles=""
  listItemStyle={null}
  listType="ul"
  theme={
    Object {
      "backgroundColorDisabled": "grey",
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "cautionBackgroundColor": "#fffdf3",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "deviceType": "desktop",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "minHeightHeaderDesktop": 76,
      "minHeightHeaderMobile": 113,
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zFilter": 100,
        "zFilterActions": 400,
        "zFilterList": 150,
        "zFlyout": 200,
        "zLoader": 1000,
        "zMenu": 800,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zPlacesAutocomplete": 1100,
        "zSticky": 700,
      },
    }
  }
/>
`;

exports[`List should render correctly for mobile 1`] = `
.c0 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

<List
  className="c0"
  defaultStyles={false}
  elementRef={null}
  inheritedStyles=""
  listItemStyle={null}
  listType="ul"
  theme={
    Object {
      "backgroundColorDisabled": "grey",
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "cautionBackgroundColor": "#fffdf3",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "deviceType": "mobile",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "minHeightHeaderDesktop": 76,
      "minHeightHeaderMobile": 113,
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zFilter": 100,
        "zFilterActions": 400,
        "zFilterList": 150,
        "zFlyout": 200,
        "zLoader": 1000,
        "zMenu": 800,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zPlacesAutocomplete": 1100,
        "zSticky": 700,
      },
    }
  }
/>
`;
